sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/RocksDBStateStoreProvider.scala

```diff
@@ -195,6 +195,12 @@ private[sql] class RocksDBStateStoreProvider
     new RocksDBStateStore(version)
   }
 
+  override def getReadStore(version: Long): StateStore = {
```

Comment:
Hmm, how was this working before this change for RocksDB ?

Reply:
It loads a write state store, but the load doesn't consume extra resources. We make the change here because now we create the changelog writer in load.